src/core/scope.c

```diff
@@ -428,8 +428,9 @@ static void scope_sigchld_event(Unit *u, pid_t pid, int code, int status) {
         unit_tidy_watch_pids(u, 0, 0);
         unit_watch_all_pids(u);
 
-        /* If the PID set is empty now, then let's finish this off */
-        if (set_isempty(u->pids))
+        /* If the PID set is empty now, then let's finish this off
+           (On unified we use proper notifications) */
```

Comment:
Nice catch. That can also be triggered by "echo 3 > /proc/sys/vm/drop_caches". The notification code was incorrectly depending on dentries sticking around and failed to generate notifications if the target dentries were reclaimed. I posted a fix patch.

http://lkml.kernel.org/g/20160617215117.GR3262@mtj.duckdns.org

Thanks a lot!